kv/txn_coord_sender.go

```diff
@@ -347,61 +304,84 @@ func (tc *TxnCoordSender) maybeBeginTxn(header *proto.RequestHeader) {
 //
 // On success, and if the call is part of a transaction, the affected
 // key range is recorded as live intents for eventual cleanup upon
-// transaction commit. Upon successful txn commit, initiates cleanup
-// of intents.
-func (tc *TxnCoordSender) sendOne(ctx context.Context, call proto.Call) {
+// transaction commit, in which case they are included in the
+// EndTransaction request sent by the client.
+func (tc *TxnCoordSender) SendBatch(ctx context.Context, ba *proto.BatchRequest) (*proto.BatchResponse, error) {
+	tc.maybeBeginTxn(ba)
+	ba.CmdID = ba.GetOrCreateCmdID(tc.clock.PhysicalNow())
 	var startNS int64
-	header := call.Args.Header()
-	trace := tracer.FromCtx(ctx)
+
+	// This is the earliest point at which the request has a ClientCmdID and/or
+	// TxnID (if applicable). Begin a Trace which follows this request.
+	trace := tc.tracer.NewTrace(ba)
+	defer trace.Finalize()
+	// TODO(tschottdorf): always "Batch"
+	defer trace.Epoch(fmt.Sprintf("sending %s", ba.Method()))()
```

Comment:
Yeah, the to-do is for figuring out whether it should be something else.
On Aug 27, 2015 19:53, "Tamir Duberstein" <notifications@github.com> wrote:

> In kv/txn_coord_sender.go
> <https://github.com/cockroachdb/cockroach/pull/2141#discussion_r38124986>:
>
> >  	var startNS int64
> > -	header := call.Args.Header()
> > -	trace := tracer.FromCtx(ctx)
> > +
> > +	// This is the earliest point at which the request has a ClientCmdID and/or
> > +	// TxnID (if applicable). Begin a Trace which follows this request.
> > +	trace := tc.tracer.NewTrace(ba)
> > +	defer trace.Finalize()
> > +	// TODO(tschottdorf): always "Batch"
> > +	defer trace.Epoch(fmt.Sprintf("sending %s", ba.Method()))()
>
> it's already the case that this is always "batch"
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/2141/files#r38124986>.
>